### PR TITLE
Add space detection

### PR DESCRIPTION
**Pull request by folksgl**
* Lots of readme improvements
* Added ability to specify config file via "-config" flag
* Added ability to view current loaded config via "/config" endpoint
* Added CF Space SSH Access misconfiguration detection
* Added tests for resource_config.go
* Modified structure of config file
* Removed config_viewer files that were caching CF data for no reason
* Removed unused dependencies

### CODE REVIEW

File: drift_detector.go [hunk 1]
```diff
@@ -2,57 +2,142 @@ package main
...
-	totalUnknownApps.Set(float64(unknownApps))
+	successfulSpaceChecks.Inc()
+}
+
+// App difference returns a slice of unknown app names.
+// Given two AppEntry slices, returns the set difference of the two
+// slices (a - b). This can be logically used as follows:
+// unknownApps = (the set of deployed apps) - (the set of valid apps) OR
+// missingApps = (the set of valid apps) - (the set of deployed apps)
+func appDifference(deployed, valid []AppEntry) (diff []string) {
```
Comment (dzaslavskiy) [inline drift_detector.go:123]: Seems like the args are miss-named on this function since it used twice in opposite inputs

File: README.md [hunk 2]
```diff
@@ -19,15 +24,103 @@ Resources are checked on an opt-in model, meaning if you
...
+## Running Watchtower
+Watchtower can be run from anywhere that is able to hit your cloud foundry api.
+To run, either download a pre-compiled binary from the [releases](https://github.com/18F/watchtower/releases)
+page, or compile the go source yourself using `go build`.
+
+### CLI Arguments
+
+| Argument | Description |
+| --- | --- |
+| `-config` | Path to the configuration file. Default is "config.yaml" in the current directory |
```
Comment (dzaslavskiy) [inline README.md:36]: single '-' ?

**Review by dzaslavskiy — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (folksgl) [reply README.md:36]: Yes, a single dash is the correct form. See https://pkg.go.dev/flag#hdr-Command_line_flag_syntax

File: drift_detector.go (hunk 1 above)
Comment (folksgl) [reply drift_detector.go:123]: Agreed. How does the following seem?
```suggestion
func appDifference(a, b []AppEntry) (diff []string) {
```

File: README.md (hunk 2 above)
Comment (dzaslavskiy) [reply README.md:36]: I see Go thinks it is special. Well at least: "One or two minus signs may be used; they are equivalent."

**Review by dzaslavskiy — APPROVED** [review]

**Pull request merged**